File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(strand_a, strand_b)
    raise ArgumentError if strand_a.length != strand_b.length
    strand_a = strand_a.split('')
    strand_b = strand_b.split('')

    difference = 0
    strand_a.each_with_index do |base, index|
      difference += 1 if base != strand_b[index]
    end
    difference
  end
end

File: elixir/hamming/hamming.exs
defmodule Hamming do
  @spec hamming_distance([char], [char]) :: non_neg_integer
  def hamming_distance(strandA, strandB) do
    { :ok, distance(Enum.zip(strandA, strandB) ) }
  end

  def distance(strands) do
    Enum.count(strands, fn({a, b}) -> a != b end)
  end
end
